The hang is fixed for a job which also had some non unicode characters showing it is safe to remove the try, except for UnicodeDecodeErrors:

```
2023-09-14T15:15:56.274434066Z stderr F + mkdir -p /storage/test-reports-269-default/log_tensorflow2-IPU-POD4-ogb-training_2023-09-14-15_15_56
2023-09-14T15:15:56.27675755Z stderr F + TEST_CONFIG_FILE=/notebooks/.gradient/notebook-tests.yaml
2023-09-14T15:15:56.276775063Z stderr F + cd /notebooks/
2023-09-14T15:15:56.27677883Z stderr F + echo 'PAPERSPACE-AUTOMATED-TESTING: starting platform_assessment testing'
2023-09-14T15:15:56.276807233Z stderr F + python -m examples_utils platform_assessment --spec /notebooks/.gradient/notebook-tests.yaml --benchmark ogb-training --progress-trace-period 60 --log-dir /storage/test-reports-269-default/log_tensorflow2-IPU-POD4-ogb-training_2023-09-14-15_15_56 --gc-monitor --cloning-directory /tmp/clones --additional-metrics
2023-09-14T15:15:56.276808878Z stdout F PAPERSPACE-AUTOMATED-TESTING: starting platform_assessment testing
2023-09-14T15:15:57.197046108Z stderr F [INFO] 2023-09-14 15:15:57: Logging directory: '/storage/test-reports-269-default/log_tensorflow2-IPU-POD4-ogb-training_2023-09-14-15_15_56'
2023-09-14T15:15:57.197196311Z stderr F [INFO] 2023-09-14 15:15:57: Running benchmark suite: ''
2023-09-14T15:15:57.208408486Z stderr F [INFO] 2023-09-14 15:15:57: ----------------------------------------
2023-09-14T15:15:57.208423775Z stderr F [INFO] 2023-09-14 15:15:57: Preparing environment for 'ogb-training'
2023-09-14T15:15:57.208558999Z stderr F [INFO] 2023-09-14 15:15:57: Install python requirements
2023-09-14T15:17:46.915917423Z stderr F Benchmark elapsed time: 0:00:02 (2 seconds) > Benchmark elapsed time: 0:00:04 (4 seconds) => Benchmark elapsed time: 0:00:06 (6 seconds) ==> Benchmark elapsed time: 0:00:08 (8 seconds) ===> Benchmark elapsed time: 0:00:10 (10 seconds) ====> Benchmark elapsed time: 0:00:12 (12 seconds) <====> Benchmark elapsed time: 0:00:14 (14 seconds) <==== Benchmark elapsed time: 0:00:16 (16 seconds) <=== Benchmark elapsed time: 0:00:18 (18 seconds) <== Benchmark elapsed time: 0:00:20 (20 seconds) <= Benchmark elapsed time: 0:00:22 (22 seconds) < Benchmark elapsed time: 0:00:24 (24 seconds) Benchmark elapsed time: 0:00:26 (26 seconds) > Benchmark elapsed time: 0:00:28 (28 seconds) => Benchmark elapsed time: 0:00:30 (30 seconds) ==> Benchmark elapsed time: 0:00:32 (32 seconds) ===> Benchmark elapsed time: 0:00:34 (34 seconds) ====> Benchmark elapsed time: 0:00:36 (36 seconds) <====> Benchmark elapsed time: 0:00:38 (38 seconds) <==== Benchmark elapsed time: 0:00:40 (40 seconds) <=== Benchmark elapsed time: 0:00:42 (42 seconds) <== Benchmark elapsed time: 0:00:44 (44 seconds) <= Benchmark elapsed time: 0:00:46 (46 seconds) < Benchmark elapsed time: 0:00:48 (48 seconds) Benchmark elapsed time: 0:00:50 (50 seconds) > Benchmark elapsed time: 0:00:52 (52 seconds) => Benchmark elapsed time: 0:00:54 (54 seconds) ==> Benchmark elapsed time: 0:00:56 (56 seconds) ===> Benchmark elapsed time: 0:00:58 (58 seconds) ====> Benchmark elapsed time: 0:01:00 (60 seconds) <====> Benchmark elapsed time: 0:01:02 (62 seconds) <==== Benchmark elapsed time: 0:01:04 (64 seconds) <=== Benchmark elapsed time: 0:01:06 (66 seconds) <== Benchmark elapsed time: 0:01:08 (68 seconds) <= Benchmark elapsed time: 0:01:10 (70 seconds) < Benchmark elapsed time: 0:01:12 (72 seconds) Benchmark elapsed time: 0:01:14 (74 seconds) > Benchmark elapsed time: 0:01:16 (76 seconds) => Benchmark elapsed time: 0:01:18 (78 seconds) ==> Benchmark elapsed time: 0:01:20 (80 seconds) ===> Benchmark elapsed time: 0:01:22 (82 seconds) ====> Benchmark elapsed time: 0:01:24 (84 seconds) <====> Benchmark elapsed time: 0:01:26 (86 seconds) <==== Benchmark elapsed time: 0:01:28 (88 seconds) <=== Benchmark elapsed time: 0:01:30 (90 seconds) <== Benchmark elapsed time: 0:01:32 (92 seconds) <= Benchmark elapsed time: 0:01:34 (94 seconds) < Benchmark elapsed time: 0:01:36 (96 seconds) Benchmark elapsed time: 0:01:38 (98 seconds) > Benchmark elapsed time: 0:01:40 (100 seconds) => Benchmark elapsed time: 0:01:42 (102 seconds) ==> Benchmark elapsed time: 0:01:44 (104 seconds) ===> Benchmark elapsed time: 0:01:46 (106 seconds) ====> Benchmark elapsed time: 0:01:48 (108 seconds) <====>
2023-09-14T15:17:46.915973949Z stderr F [INFO] 2023-09-14 15:17:46: Logs at: /storage/test-reports-269-default/log_tensorflow2-IPU-POD4-ogb-training_2023-09-14-15_15_56/output.log
2023-09-14T15:17:46.916096971Z stderr F [INFO] 2023-09-14 15:17:46: Running ogb-training
2023-09-14T15:17:46.923713408Z stderr F [INFO] 2023-09-14 15:17:46: original cmd = 'python3 -m examples_utils.benchmarks.notebook_utils gps++_training.ipynb . --timeout 16000'
2023-09-14T15:17:46.923744576Z stderr F [INFO] 2023-09-14 15:17:46: Cleaning and modifying command if required...
2023-09-14T15:17:46.923752772Z stderr F [INFO] 2023-09-14 15:17:46: new cmd = 'python3 -m examples_utils.benchmarks.notebook_utils gps++_training.ipynb . --timeout 16000'
2023-09-14T15:17:46.923756749Z stderr F [INFO] 2023-09-14 15:17:46: Running with the following 3 ADDITIONAL ENV variables:
2023-09-14T15:17:46.923760977Z stderr F [INFO] 2023-09-14 15:17:46: POPLAR_LOG_LEVEL=INFO
2023-09-14T15:17:46.923765535Z stderr F [INFO] 2023-09-14 15:17:46: POPART_LOG_LEVEL=INFO
2023-09-14T15:17:46.92376808Z stderr F [INFO] 2023-09-14 15:17:46: TF_CPP_VMODULE=poplar_compiler=1
2023-09-14T15:17:46.923775745Z stderr F [INFO] 2023-09-14 15:17:46: cwd = '/notebooks/ogb-competition'
2023-09-14T15:17:46.924096399Z stderr F [INFO] 2023-09-14 15:17:46: Datasets directory: '/tmp/dataset_cache'
2023-09-14T15:17:46.92415537Z stderr F [INFO] 2023-09-14 15:17:46: poprun not called, assuming this is a single-host, single-instance benchmark.
2023-09-14T15:17:46.924183983Z stderr F [INFO] 2023-09-14 15:17:46: Install python requirements
2023-09-14T15:17:47.237034849Z stderr F Running command git clone --filter=blob:none -q https://github.com/graphcore/graphcore-cloud-tools /tmp/pip-install-8k4wyhgw/graphcore-cloud-tools_b8559907f914449a884edbc4e12f945f
2023-09-14T15:17:53.90961151Z stderr F Running command git checkout -q 0af1fc805cfb8f93fe2e2407e32bfeff2dce1f55
2023-09-14T15:17:59.605227555Z stderr F [INFO] 2023-09-14 15:17:59: Start test: 2023-09-14 15:17:59.604932
2023-09-14T15:40:20.623521403Z stderr F Benchmark elapsed time: 0:01:01 (61 seconds) > Benchmark elapsed time: 0:02:02 (122 seconds) => Benchmark elapsed time: 0:03:03 (183 seconds) ==> Benchmark elapsed time: 0:04:04 (244 seconds) ===> Benchmark elapsed time: 0:05:05 (305 seconds) ====> Benchmark elapsed time: 0:06:06 (366 seconds) <====> Benchmark elapsed time: 0:07:07 (427 seconds) <==== Benchmark elapsed time: 0:08:08 (488 seconds) <=== Benchmark elapsed time: 0:09:09 (549 seconds) <== Benchmark elapsed time: 0:10:10 (610 seconds) <= Benchmark elapsed time: 0:11:11 (671 seconds) < Benchmark elapsed time: 0:12:12 (732 seconds) Benchmark elapsed time: 0:13:13 (793 seconds) > Benchmark elapsed time: 0:14:14 (854 seconds) => Benchmark elapsed time: 0:15:15 (915 seconds) ==> Benchmark elapsed time: 0:16:16 (976 seconds) ===> Benchmark elapsed time: 0:17:17 (1037 seconds) ====> Benchmark elapsed time: 0:18:18 (1098 seconds) <====> Benchmark elapsed time: 0:19:19 (1159 seconds) <==== Benchmark elapsed time: 0:20:20 (1220 seconds) <=== Benchmark elapsed time: 0:21:21 (1281 seconds) <==[INFO] 2023-09-14 15:40:20: Selector did not pick any files to explore, and subprocess has exited. Terminating.
2023-09-14T15:40:40.651792853Z stderr F Benchmark elapsed time: 0:22:22 (1342 seconds) <=[WARNING] 2023-09-14 15:40:40: I/O Thread failed to communicate with process at the end of benchmark, timing out to avoid lock up. Benchmark logs may be truncated.
2023-09-14T15:40:43.763511297Z stderr F
2023-09-14T15:40:43.763580698Z stderr F [INFO] 2023-09-14 15:40:43: End test: 2023-09-14 15:40:43.761773
2023-09-14T15:40:43.763983425Z stderr F [INFO] 2023-09-14 15:40:43: Total runtime: 1364.156841 seconds
2023-09-14T15:40:43.764459843Z stderr F [ERROR] 2023-09-14 15:40:43: Benchmark ERROR, exited with code: (1). Please check logs for more information.
```

See full results:

[log_tensorflow2-IPU-POD4-ogb-training_2023-09-14-15_15_56.tar.gz](https://github.com/graphcore/examples-utils/files/12610993/log_tensorflow2-IPU-POD4-ogb-training_2023-09-14-15_15_56.tar.gz)

--- a/.github/workflows/pytest.yaml
+++ b/.github/workflows/pytest.yaml
@@ -13,10 +13,6 @@ jobs:
       VIRTUAL_ENV: /some/fake/venv/path
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -22,7 +22,7 @@ repos:
       - id: clang-format
         args: [-i]
   - repo: https://github.com/graphcore/examples-utils
-    rev: latest_stable
+    rev: v3.4
     hooks:
     - id: copyright-header-check
       args: [--amend]

